main.cpp

```diff
@@ -73,26 +79,557 @@ write 3 UDTs below that EACH have:
+    {
+        bloodDonated += 100;
+        girlfriend.amountOfBlood -= 100;
+        std::cout << "still donating: " << bloodDonated << "/500... hang in there.\n";
+    }
+    std::cout << "all done, go home\n";
+}
+
+void Human::printHumanVars()
+{
+    std::cout << "ageInYears: " << ageInYears << " \n";
+    std::cout << "nameOfHuman: " << nameOfHuman << " \n";
+    std::cout << "ethnicity: " << ethnicity << "\n";
+    std::cout << "dateOfBirth:  " << dateOfBirth << "\n";
+    std::cout << "bloodType: " << bloodType << "\n\n"; 
+}
+
 /*
  new UDT 4:
+ ===============================================================
  with 2 member functions
  */
+struct CatShelter
+{
+    PetCat msDoris;
+    PetCat::CatCollar catCollar;
+    int roomForCat {1000};
+
+    CatShelter();
+    ~CatShelter();
+
+    int AdmitNewCat(PetCat newCat, PetCat::CatCollar newCollar);
+
+    int getCatAdopted(PetCat newCat, Human catAdopter);
+};
+
+CatShelter::CatShelter()
+{
+    std::cout << "CatShelter is being constructed\n";
+}
+
+CatShelter::~CatShelter()
+{
+    std::cout << "CatShelter deconstructed\n";
+    msDoris.printPetCatVars();
+}
+
+int CatShelter::AdmitNewCat(PetCat newCat, PetCat::CatCollar newCollar)
+{   
+    if(newCat.hasVaccines == false)
+    {
+        std::cout << "administering all vacines\n";
+        newCat.hasVaccines = true;
+    }
+
+    if(newCollar.idNumber == roomForCat)
+    {
+        roomForCat -= 1;
+    }
+    return roomForCat; 
+}
+
+int CatShelter::getCatAdopted(PetCat newCat, Human catAdopter)
+{
+    int ageRequirement = 18;
+    int newOwnersAge = catAdopter.ageInYears;
+    std::string catsNewOwner = catAdopter.nameOfHuman;
+        
+    if(newOwnersAge < ageRequirement)
+    {
+        std::cout << "you are too young to legally adopt this cat\n";
+    }
+    else
+    {
+        std::cout << catsNewOwner << "thank you for adopting " << newCat.nameOfPetCat << "\n";
+        roomForCat += 1; 
+    }
+    return roomForCat;
+}
 
 /*
  new UDT 5:
+ ===============================================================
  with 2 member functions
  */
+struct Hospital
+{
+    Human pateint;
+    Human::HealthStatus goodHealth;
+    int numOfPatients;
+
+    Hospital();
+    ~Hospital();
 
+    bool giveXray(Human currentUser, Human::HealthStatus currentStatus);
+
+    bool recomendMedication(Human injuredPatient, Human::HealthStatus injuredStatus);
+};
+
+Hospital::Hospital() :
+numOfPatients(1000)
+{
+     std::cout << "Hospital is being constructed\n";  
+}
+
+Hospital::~Hospital()
+{
+    std::cout << "Hospital is being constructed\n";
+    pateint.printHumanVars();
+}
+
+bool Hospital::giveXray(Human currentPatient, Human::HealthStatus currentStatus)
+{
+    std::cout << currentPatient.nameOfHuman << " Are you experiencing any pain?\n";
+        
+    if (currentStatus.experiencingPain == true)
+    {
+        return true;
+    }
+    else
+    {
+        return false;
+    }
```

Comment:
If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
You do not need the `else` block.  Here's why:
`return` statements are the last thing that **ever happens** in a function.
When the `if()` expression is evaluated as `true` and the block is entered, the `return aValue;` is the last line of code to be executed before the function exits.
but when the `if()` expression evaluates as `false` and the block is **NOT** entered, the next thing that happens is `return anotherValue;`
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
that means you do **NOT** need the `else` block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    
    return anotherValue;
}
```
- [ ] revise